### ISSUE TITLE
where is 2M audioset data and pretrain_audioset2M.sh?

### ISSUE THREAD

**Issue by JHjang223** [facebookresearch/AudioMAE#21]
Thank you meta for your hard work on the audioMAE implementation.
I want to train with 2M data, but in fact, audioset is only releasing features, so I couldn't get the data. I was finally able to get 20k data from another website. Where do I download the 2M data and I can't find pretrain_audioset2M.sh. Check please.

**Comment by Gariscat** [comment]
Same issue...... I checked the website and also only found the features instead of the original waveforms. How should we get the raw data or the raw data is not released at all?

**Comment by Jingerjia** [comment]
My stupid solution is:
Download the html of the class, and you'll find each video has it's youtube-id, start time, end time, and labels.
Then we can download every video we need by analyzing the html of the classes.
Good luck!

**Comment by IvanBirkmaier** [comment]
You can also use the .wav data which is provided by Huggingface: https://huggingface.co/datasets/confit/audioset-full or Baidu:  https://pan.baidu.com/s/13WnzI1XDSvqXZQTS-Kqujg, password: 0vc2 (source: https://github.com/qiuqiangkong/audioset_tagging_cnn). 

In the Hugginface dataset (eval) there is one broken file: ID YmW3... (if i remember right) delete this one it can cause headach :D
After downloading the data you have to create an train and eval json like they did in AST (https://github.com/YuanGongND/ast) (see egs/audioset/datafiles/sample_...) don't forget you just need audio an label!!! 

must look like this:

        {
            "wav": "your path to wav file (doesn't have to be .flac file -> torchaudio supports both)",
            "labels": "/m/068hy,/m/07q6cd_,/m/0bt9lr,/m/0jbk"
        },
        
The label mapping for the wav-files/data can be done with the https://github.com/audioset/ontology and the provided CSV files (balanced_train_segments.csv, etc.) given on Audioset website: https://research.google.com/audioset/download.html

Good Luck

**Comment by ZitongLan** [comment]
Hi @IvanBirkmaier  I have tried to download the dataset from the hugging face. However, I found that some of the data point is missing ( exists in train_all.json, but not in the dataset) do you have any idea of this?

**Comment by IvanBirkmaier** [comment]
Hey @ZitongLan this is because the YT videos used to create audioset were removed from YT over time and it was therefore no longer possible to restore all the wav files of the original audio set. Have a look at my repository: https://github.com/IvanBirkmaier/Audioset/tree/master I analyzed the Huggingface data set with EDA (see readme). There you can also easily create the JSONs for the audio MAE recipie. If there is any issue with the repository feel free to commit an issue

**Comment by ZitongLan** [comment]
@IvanBirkmaier Thanks for your help! I will take a look on the repo and the dataset. If anything, I will just post issue their. Thanks!

**Comment by IvanBirkmaier** [comment]
@ZitongLan You're welcome. Fyi when using the HF dataset for reproducing the audioset recipes, the results for audio spectogram transformer (AST https://github.com/YuanGongND/ast) looked good (comparable with the paper), but for audio MAE the performance dropped. Even with the provided pretrained ckpt. But of course maybe it was just a "bad" run. - will add them into the repo
